### ISSUE TITLE
[Bug]: chrome://inspect + click on menu item + pause in debugger = whole system freeze

### ISSUE THREAD

**Issue by safinaskar** [electron/electron#35671]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

20.1.3

### What operating system are you using?

Other Linux

### Operating System Version

Debian Bullseye

### What arch are you using?

x64

### Expected Behavior

.

### Actual Behavior

Innocent debugging using chrome://inspect can freeze whole system and make whole system unusable until reboot! Here is steps to reproduce.

I have Debian Bullseye, KDE with X11 (note: X11, not Wayland), Electron 20.1.3, Chromium 104.0.5112.79.

Here is my `package.json`:
```js
{
  "name": "pr",
  "version": "0.1.0",
  "main": "main.js",
  "scripts": {
    "start": "electron ."
  },
  "devDependencies": {
    "electron": "^20.0.2"
  }
}
```

Here is my `main.js`:
```js
"use strict";

const { app, BrowserWindow, BrowserView, dialog, clipboard, ipcMain, Menu, MenuItem } = require("electron");
const path = require("path");

(async () => {
  await app.whenReady();

  const window = new BrowserWindow({});
  {
    const menu = new Menu;
    menu.append(new MenuItem({
      label: "Menu",
      submenu: [
        {
          label: "Item",
          click: () => {
            console.log("a");
            console.log("b");
            console.log("c");
          }
        }
      ]
    }));
    Menu.setApplicationMenu(menu);
  }
})();
```

Start the app using `./node_modules/electron/dist/electron --inspect .`. Then open `chrome://inspect` in Chromium, from there open devtools. In devtools open tab "Sources", then open file `main.js`. Then set breakpoint at `console.log("a")`. Then switch to electron app, click at menu "Menu" and then click to menu item "Item". At this moment devtools should pause at breakpoint `console.log("a")`. And in this point you will discover that you cannot switch between windows using your mouse anymore! Starting from this moment attempting to click to titlebar of any window to switch to it will fail! The system becomes unusable!

**Comment by VerteDinde** [comment]
@safinaskar Is this behavior unique to Electron 20? Can you please try this with Electron 19 and/or Electron 18 as well?

**Comment by safinaskar** [comment]
I tried Electron 19.0.17 and 18.3.13. They are buggy, too

**Comment by safinaskar** [comment]
**Workaround** for other people in the same situation! So, if your system got freezed, simply press F8 ("Resume execution") while devtools still active. This should continue execution and unfreeze your system

**Comment by safinaskar** [comment]
Bug reproduces differently on KDE Wayland. The system stays usable, switching between apps works, Konsole (sic!) continues to work. But mouse stops to work with Chromium window. So, overall Wayland performs significantly better. Also that F8 workaround works, too

**Comment by codebytere** [comment]
@safinaskar does this occur in Chrome itself?

**Comment by safinaskar** [comment]
@codebytere , you mean I should try Chrome instead of Chromium? Or you mean somehow reproduce this bug on Chrome/Chromium without Electron?

**Comment by codebytere** [comment]
@safinaskar yes - if this happens when you use Google Chrome, _without Electron_ then that indicates it's an upstream issue and should be reported there and not on our tracker.

**Comment by safinaskar** [comment]
I tried to reproduce this bugs on Chromium alone, without Electron, using Chromium extensions and contextMenus ( https://developer.chrome.com/docs/extensions/reference/contextMenus ). Here is Chromium extension I created for this. This is `manifest.json`:
```json
{
  "name": "test",
  "version": "0.1",
  "manifest_version": 2,
  "permissions": ["contextMenus"],
  "background": {
    "scripts": ["background.js"]
  }
}
```
And here is `background.js`:
```js
"use strict";

chrome.runtime.onInstalled.addListener(() => {
  chrome.contextMenus.create({ id: "a", title: "a" });
});

chrome.contextMenus.onClicked.addListener(() => {
  console.log("a");
  console.log("b");
  console.log("c");
});
```

If you want I can tell in more details how I tried to reproduce this bug. **Surprisingly, bug was not reproduced.** (In Chromium 104.0.5112.79.) I suspect this is because Chromium closes menu before executing user-provided handler, so, I think Electron should do the same. (In Electron menu remained to be visible when I was stopped in debugger, but in Chromium menu disappeared.)

Also, I think this bug is caused by fundamental characteristic of X11: X11 doesn't know anything about menus, they are simply implemented as windows, which grab all mouse events. Thus stopping any app (with menu opened) in debugger disables mouse. As I said, I think this could be fixed by closing menu before running handler.

I don't know why this bug reproduces in Wayland. I suspect this is because Chromium and Electron are run in X11-compatibility mode, i. e. using XWayland (they both show up in `wmctrl -l` output)

**Comment by safinaskar** [comment]
Owners of this repo, please, disable this annoying bot. See also: https://drewdevault.com/2021/10/26/stalebot.html

**Comment by safinaskar** [comment]
bump

**Comment by safinaskar** [comment]
https://drewdevault.com/2021/10/26/stalebot.html
